docs/src/architecture/execution.md

```diff
@@ -1,83 +1,59 @@
+To reduce the required space on the Ethereum blockchain, transaction proofs are being aggregated into batches. This can happen in parallel by different machines that need to verify several proofs using the Miden VM and thus creating a proof thereof. Verifying a STARK proof within the VM is relatively efficient - but it is still a pretty costly operation (we aim for 2^16 cycles).
 
-We can also take the batching approach a step further and add more recursion layers (i.e., batches of batches) - but this will complicate both the ag kernel and the os kernel quite a bit. So, my thinking is that at least initially, we don't try to do that.
+## Block Production
+Several batch proofs are being aggregated together into one block. This can not happen in parallel and must be done by the Miden Operator running the Miden Node. The idea is the same, using recursive verification.
 
-Another interesting benefit of batching is that we may be able to use it to support proof generation on resource-constraint devices. One of the reasons why proof generation for transactions will be expensive is that to do recursive proof aggregation we need to build lower-level proofs using arithmetization-friendly hash functions (i.e., Rescue Prime in our case). These hash functions are not very efficient (i.e., probably 30x less efficient than BLAKE3) and will dominate proof generation costs.
 
-However, with batching, we can allow the following:
+# State progress 
+At the beginning Miden will have a centralized operator running a Miden Node.
 
-1. Clients generate their tx proofs using BLAKE3 or SHA256 hash functions - which should be significantly faster than using Rescue Prime or similar.
-2. Batch producer verifies these proofs and outputs a batch proof which uses Rescue Prime as the hash function.
-3. Block producer aggregates batches as before. The fact that some tx proofs were built using non-arithmetization-friendly hashes is transparent to them.
-
-Recursively verifying proofs built using BLAKE3/SHA256 would be quite a bit more expensive - so, such transactions would need to include higher fee - but there could be use cases where this may be justified.
-
-## State progress 
-For simplicity, let's imagine a simple centralized model:
-* The Miden Node collects transactions from users.
-* The Miden Node puts collected transactions into blocks.
-* The blocks get submitted to Ethereum L1, and once a block is included in the L1 chain, the rollup chain is assumed to have moved to the next state.
+* The Miden Node collects proofs or transactions to be proven from the users.
+* The Miden Node aggregates collected transaction proofs into batches using recursive verification.
+* The Miden Node aggregates batch proofs into blocks.
+* The blocks get submitted to Ethereum, and once a block is included in the L1 chain, the rollup chain is assumed to have moved to the next state. (not specified yet)
```

Comment:
Do we want to mention that we will aggregate block proofs in "checkpoints proofs" such that we only verify a single proof that attests the to the aggregated state transition of a number of blocks.

Reply:
Yeah, that's actually a good idea. That was what I was referrfing in our last Team sync when I mentioned "epochs". 

I suggest to not mention it now and wait until we have specified how we want to do it.